src/particles/elements/Sol.H

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_SOL_H
+#define IMPACTX_SOL_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/beamoptic.H"
+#include "mixin/beamoptic.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct Sol
```

Comment:
Do we want to name this class `Solenoid` as well?

Reply:
Sure; I think either name is fine.  The abbreviation Sol is in common use for a solenoid, so I don't think there is much chance for confusion.